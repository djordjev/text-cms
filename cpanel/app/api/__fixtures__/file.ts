import { TEST_EDITOR } from '~/utils/test';

export const FILE_CONTENT = [
  {
    id: '1',
    name: 'First Variation',
    condition: [[['a', '=', '1']]],
    text: TEST_EDITOR
  },
  {
    id: '2',
    name: 'Second Variation',
    condition: [
      [
        ['b', '=', '1'],
        ['c', '>', '12']
      ],
      [['w', '=', 'true']]
    ],
    text: TEST_EDITOR
  }
];

export const FILE_VARIATION = JSON.stringify(FILE_CONTENT);
