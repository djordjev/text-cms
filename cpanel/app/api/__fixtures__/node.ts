import { FsNode } from '@prisma/client';

const newNode = (prototype?: Partial<FsNode>, children?: Partial<FsNode>[]) => {
  // eslint-disable-next-line
  const result: any = {
    createdAt: prototype?.createdAt ?? new Date(2024, 2, 6, 12, 12, 12, 12),
    id: prototype?.id ?? 1,
    name: prototype?.name ?? 'random_name',
    parentId: prototype?.parentId ?? 1001,
    path: prototype?.path ?? '/randompath',
    typeId: prototype?.typeId ?? 1
  };

  if (children) {
    result.children = [];

    for (const child of children) {
      const newChild = {
        createdAt: child?.createdAt ?? new Date(2024, 2, 6, 12, 12, 12, 12),
        id: child?.id ?? 1,
        name: child?.name ?? 'random_name',
        parentId: child?.parentId ?? 1001,
        path: child?.path ?? '/randompath',
        typeId: child?.typeId ?? 1
      };

      result.children.push(newChild);
    }
  }

  return result;
};

export { newNode };
