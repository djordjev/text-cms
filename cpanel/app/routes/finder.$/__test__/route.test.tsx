import { createRemixStub } from '@remix-run/testing';
import { fireEvent, render, screen } from '@testing-library/react';

import { newNode } from '~/api/__fixtures__/node';

import Finder from '../route';

test('finder.$ route', async () => {
  const createComponent = () => {
    const RemixStub = createRemixStub([
      {
        path: '/finder',
        Component: () => <Finder />,
        loader: () => {
          return [
            newNode({ id: 20, name: 'Folder1', typeId: 1 }),
            newNode({ id: 21, name: 'File1.txt', typeId: 2 })
          ];
        }
      }
    ]);

    return <RemixStub initialEntries={['/finder']} />;
  };

  const { container } = render(createComponent());

  // Awaits for breadcrumbs
  expect(await screen.findByText('Home')).toBeInTheDocument();

  // Renders file and folder
  expect(screen.getByRole('img', { name: 'Folder1' })).toBeInTheDocument();
  expect(screen.getByRole('img', { name: 'File1.txt' })).toBeInTheDocument();

  // Matches screenshot
  expect(container.childNodes).toMatchSnapshot();

  // Right click - delete
  fireEvent.contextMenu(screen.getByText('File1.txt'));
  expect(screen.getByText('Delete')).toBeInTheDocument();

  // Right click - background
  fireEvent.contextMenu(screen.getByTestId('background'));
  expect(screen.getByText('New Folder')).toBeInTheDocument();
  expect(screen.getByText('New File')).toBeInTheDocument();
});
