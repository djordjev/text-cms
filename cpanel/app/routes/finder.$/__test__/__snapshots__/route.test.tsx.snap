// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`finder.$ route 1`] = `
NodeList [
  <div
    class="u-flex u-flex-col u-flex-grow"
  >
    <div
      class="u-flex u-items-center"
      style="padding-top: 0.5rem; padding-bottom: 0.5rem;"
    >
      <svg
        class="u-inline"
        fill="none"
        height="18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 12l-2 0l9 -9l9 9l-2 0"
        />
        <path
          d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7"
        />
        <path
          d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6"
        />
      </svg>
      <span
        class="u-align-middle u-text-sm"
      >
        Home
      </span>
    </div>
    <div
      class="u-mt-5x u-flex u-flex-grow u-flex-wrap"
      data-testid="background"
    >
      <div
        class="u-self-start u-w-1z u-max-w-1z"
      >
        <a
          class="u-flex u-flex-col u-items-center u-group u-w-1z u-max-w-1z"
          href="/finder/Folder1"
        >
          <img
            alt="Folder1"
            height="60"
            loading="lazy"
            src="/images/folder.png"
            width="60"
          />
          <span
            class="group-hover:u-underline u-max-w-full u-text-ellipsis u-overflow-hidden u-text-nowrap"
          >
            Folder1
          </span>
        </a>
      </div>
      <div
        class="u-self-start u-w-1z u-max-w-1z"
      >
        <a
          class="u-flex u-items-center u-flex-col u-group"
          href="/editor/21"
        >
          <img
            alt="File1.txt"
            height="60"
            loading="lazy"
            src="/images/file.png"
            width="60"
          />
          <span
            class="group-hover:u-underline u-max-w-full u-text-ellipsis u-overflow-hidden u-text-nowrap"
          >
            File1.txt
          </span>
        </a>
      </div>
    </div>
  </div>,
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />,
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />,
]
`;
